Assets/Feature01/Scripts/Health/HealthManager.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HealthManager : MonoBehaviour
+{
+
+
+    public static HealthManager Instance { get; private set; }
+
+    private void Awake()
+    {
+            Instance = this;       
+    }
+
+
+    public float AddHealth(float currentHealth,float powerUp,float maxHealth)
+    {
+       return (currentHealth+powerUp)>maxHealth? maxHealth:(currentHealth+powerUp);
+
+    }
+
+
+
+    public float DamageHealth(float currentHealth,float damage)
+    {
+        return (currentHealth - damage) < 0 ? 0 : currentHealth - damage;
+    }
+
+
+
+    public float CalculateCurrentHealth(PolarityManager.PlayerState oldState,PolarityManager.PlayerState newState, float _currentHealth,Dictionary<PolarityManager.PlayerState,Stats> PlayerStatesSelection)
```

Comment:
This got complicated as you didn't bring in the stats here. 
Here's my suggestion. Have two private stats variable here. 
1) Current Stats
2) Previous Stats

Switch them out when stats change. Once you do it. these methods can be simplified.